build.gradle.kts

```diff
@@ -39,20 +39,28 @@ java {
 }
 
 tasks.withType<Javadoc> {
-    exclude("/com/mongodb/hibernate/internal/**")
-    exclude("com/mongodb/hibernate/dialect/**")
-    exclude("com/mongodb/hibernate/jdbc/**")
-
     val standardDocletOptions = options as StandardJavadocDocletOptions
     standardDocletOptions.apply {
+        addBooleanOption("Werror", false)
+        // TODO-HIBERNATE-129 addStringOption("-link-modularity-mismatch", "info")
```

Comment:
We can't enable `-Werror` because of the single warning we have that can only be disabled in Java SE 18+.